src/Microsoft.AspNet.Razor/TagHelpers/TagHelperAttributeDescriptorComparer.cs

```diff
@@ -31,7 +31,8 @@ protected TagHelperAttributeDescriptorComparer()
         /// <remarks>
         /// Determines equality based on <see cref="TagHelperAttributeDescriptor.IsIndexer"/>,
         /// <see cref="TagHelperAttributeDescriptor.Name"/>, <see cref="TagHelperAttributeDescriptor.PropertyName"/>,
-        /// and <see cref="TagHelperAttributeDescriptor.TypeName"/>. Ignores
+        /// <see cref="TagHelperAttributeDescriptor.TypeName"/>, and
+        /// <see cref="TagHelperAttributeDescriptor.UsageDescriptor"/>. Ignores
```

Comment:
That assumes that both `TagHelperAttributeDescriptor`s come from the same assembly which is wrong (we don't have the assembly on the descriptor).